src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/AmazonSqsIntegrationTest.php

```diff
@@ -55,6 +56,20 @@ private function execute(string $dsn): void
 
         $this->assertEquals('{"message": "Hi"}', $encoded['body']);
         $this->assertEquals(['type' => DummyMessage::class, DummyMessage::class => 'special'], $encoded['headers']);
+
+        $this->waitUntilElapsed(seconds: 1.0, since: $messageSentAt);
+        $connection->keepalive($encoded['id']);
+        $this->waitUntilElapsed(seconds: 2.0, since: $messageSentAt);
+        $this->assertSame(0, $connection->getMessageCount(), 'The queue should be empty since visibility timeout was extended');
+        $connection->delete($encoded['id']);
+    }
+
+    private function waitUntilElapsed(float $seconds, float $since): void
```

Comment:
What about using `symfony/clock`? IMHO this should be doable and ease testing a lot

Reply:
I'm not sure this makes sense: here we're doing an integration test, which means we'd the remote server to also use a mock clock. That's not possible.